Core module update

obedjamir opened the pull request:
added composite key class for TripUserMapping
manager classes added for all entities

obedjamir committed:
Core module update
added composite key class for TripUserMapping
manager classes added for all entities

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java:
```diff
@@ -0,0 +1,49 @@
+     * Delete record by id
+     * @param id
+     */
+    public String deleteTripActivity(Long id) {
```
Return a boolean.

PinakiGhosh replied:
Create a private method which check whether a trip exists or not. Use that wherever this functionality is required

obedjamir replied:
This has been resolved

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripManager.java:
```diff
@@ -0,0 +1,49 @@
+     * @param id
+     */
+    public String deleteTrip(Long id) {
+        if(tripRepository.findById(id).isEmpty())
```
Same as previous

obedjamir replied:
This has been implemented

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripUserMappingManager.java:
```diff
@@ -0,0 +1,51 @@
+     * Insert to TripUserMapping table
+     * @param data
+     */
+    public TripUserMapping saveTripUserMapping(TripUserMapping data)
```
This should take a user id and trip id

obedjamir replied:
This method now takes userId and tripId

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripUserMappingManager.java:
```diff
@@ -0,0 +1,51 @@
+    /**
+     * Get record by trip_id and user_id
+     */
+    public Optional<TripUserMapping> getTripUserMappingById(TripUserMapping id) {
```
This is not required

obedjamir replied:
this method has been removed

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripUserMappingManager.java:
```diff
@@ -0,0 +1,51 @@
+     * Delete record by trip_id and user_id
+     * @param id
+     */
+    public String deleteTripUserMapping(TripUserMapping id) {
```
This should take a tripId and userId

obedjamir replied:
This method now takes in user_id and trip_id

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripUserMappingManager.java:
```diff
@@ -0,0 +1,51 @@
+    /**
+     * Update record
+     */
+    public TripUserMapping updateTripUserMapping(TripUserMapping tripUserMap) {
```
What will the update do?

obedjam replied:
update dose not update null attributes

obedjamir replied:
Update now does not update null attributes

PinakiGhosh replied:
TripUser mapping has only a user and tripId so what will we update?

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/TripUserCompositeKey.java:
```diff
@@ -0,0 +1,15 @@
+@AllArgsConstructor
+@NoArgsConstructor
+public class TripUserCompositeKey implements Serializable {
+    private Long trip_id;
```
tripId -> no underscore in variables

obedjamir replied:
This has been corrected

obedjamir committed:
Core module update
All requested changes implemented

obedjamir committed:
Variable name correction
Under score removed from variables

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java:
```diff
@@ -26,24 +26,34 @@ public Optional<TripActivity> getTripActivityById(Long id) {
+        return null;
+    }
+
+    private Optional<TripActivity> CheckData(Long id)
```
Function name should always start with smaller case
The name should tell you what the function does. Here the function name should be tripExists() and it should return true or false

obedjamir replied:
Still getting used to the naming convention. I have changed it to checkData

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java:
```diff
@@ -26,24 +26,34 @@ public Optional<TripActivity> getTripActivityById(Long id) {
-        if(tripActivityRepository.findById(tripActivity.getTripId()).isEmpty())
+
+        TripActivity presentData = CheckData(tripActivity.getTripId()).orElse(null);
+        if(presentData!=null)
```
When we are using optional try to not due null comparisons
Optional<TripActivity> tripOptional = = CheckData(tripActivity.getTripId());
if (tripOptional.isPresent()){
...
}

obedjamir replied:
Missed out on some parts. i have made the corrections

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripManager.java:
```diff
@@ -26,24 +26,33 @@ public Optional<Trip> getTripById(Long id) {
+        return null;
+    }
+
+    private Optional<Trip> CheckData(Long id)
```
Same as above

obedjamir replied:
This has been resolved

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripUserMappingManager.java:
```diff
@@ -14,38 +14,41 @@ public class TripUserMappingManager {
+     * @param user_id
      */
-    public TripUserMapping saveTripUserMapping(TripUserMapping data)
+    public TripUserMapping saveTripUserMapping(Long trip_id,Long user_id)
```
Dont use underscore in variable names

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/Trip.java:
```diff
@@ -44,4 +44,28 @@ public class Trip implements Serializable {
     @Column(name = "last_modified_date")
     private Date lastModifiedTime;
+
+    public void updateData(Trip trip)
```
Model classes should not have this. This should be in manager

obedjamir replied:
this method has been moved to manager class

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/TripActivity.java:
```diff
@@ -49,4 +49,24 @@ public class TripActivity implements Serializable {
     @Column(name = "last_modified_date")
     private Date lastModifiedTime;
+
+    public void updateData(TripActivity tripActivity)
```
Model classes should not have this

obedjamir replied:
Moved manager class

obedjamir committed:
Core module update
All requested changes implemented

obedjamir committed:
Changes to updateData method
updateData now returns the updated object

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java:
```diff
@@ -0,0 +1,82 @@
+        return null;
+    }
+
+    private Optional<TripActivity> checkData(Long id)
```
rename to trip exists

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java:
```diff
@@ -0,0 +1,82 @@
+     * Insert to TripActivity table
+     * @param data
+     */
+    public TripActivity saveTripActivity(TripActivity data)
```
Merge save and update into a single function.
The function flow will be
1. Check if incoming data has an id - if it has id then its an update else a create new
2. For update fetch the existing trip details and populate non null fields from new data to existing model

obedjamir replied:
This haas been implemented

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripManager.java:
```diff
@@ -0,0 +1,83 @@
+     * Insert to Trip table
+     * @param data
+     */
+    public Trip saveTrip(Trip data)
```
Same as previous

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripManager.java:
```diff
@@ -0,0 +1,83 @@
+        return null;
+    }
+
+    private Optional<Trip> checkData(Long id)
```
Rename to exists

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/UsersManager.java:
```diff
@@ -0,0 +1,84 @@
+
+    }
+
+    private Optional<Users> checkData(Long id)
```
Rename to exists

obedjamir replied:
renamed to exixts

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/TripUserMapping.java:
```diff
@@ -44,4 +45,12 @@ public class TripUserMapping implements Serializable {
     @Column(name = "last_modified_date")
     private Date lastModifiedTime;
+
+    public TripUserMapping(Long tripId, Long userId, UserRole role) {
```
Add an @AllArgsCountructor. Dont need to define explicitly

obedjamir replied:
This has been resolved

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/UsersManager.java:
```diff
@@ -0,0 +1,84 @@
+
+import java.util.Optional;
+
+public class UsersManager {
```
Need a login method here. Take an email and password as input and check if its correct. Return the user details if login is success else return null

obedjamir replied:
i have implemented user login using spring boot starter security.

obedjamir committed:
Core module updated
User login implemented using spring boot starter security
checkData renamed to exists
Save and update methods merged
